react/components/ProductImages/components/Carousel/ThumbnailSwiper.js

```diff
@@ -67,8 +68,8 @@ const ThumbnailSwiper = props => {
     thumbnailAspectRatio,
     thumbnailMaxHeight,
     displayThumbnailsArrows,
-    ...swiperProps
-  } = props
+    onThumbSwiper,
+  } = useContext(CarouselContext)
```

Comment:
Put the `useContext(CarouselContext)` near to the context, in the Index file.

```suggestion
  // The carousel hook
} =  useCarouselContext()
```

Reply:
```tsx
export function useCarouselContext() {
  const context = useContext(CarouselContext)
 
  if (context === undefined) {
    throw Error('CarouselContext should be in the react tree. Maybe you are trying to use Carousel component <link da doc aqui>')
  }

  return context
}
```

For this snippet to make sense, you will need to change the default value of context to undefined. By doing this you prevent users use context without any property.